core/src/main/java/io/questdb/griffin/engine/join/HashJoinLightRecordCursorFactory.java

```diff
@@ -177,7 +176,11 @@ public boolean hasNext() {
 
         @Override
         public long size() {
-            return -1;
+            if (size > -1) {
```

Comment:
Do we want to change behavior in the following similar cases? `SampleByInterpolateRecordCursor`, `RostiRecordCursor`, `LimitedSizeSortedLightRecordCursor`, `LatestByAllIndexedRecordCursor`, `LatestByLightRecordCursor`, `LatestByValueListRecordCursor`, `LatestByValuesIndexedFilteredRecordCursor`

Reply:
it is a bit list, i would suggest we pick this up at a later day. This has been enough of a detour already

Reply:
Sure. I'll create a GH issue for that.